instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/action_controller_subscriber_test.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+# Copyright The OpenTelemetry Authors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+require 'test_helper'
+
+require_relative '../../../../lib/opentelemetry/instrumentation/action_pack'
+require_relative '../../../../lib/opentelemetry/instrumentation/action_pack/action_controller_subscriber'
+
+describe OpenTelemetry::Instrumentation::ActionPack::ActionControllerSubscriber do
+  include Rack::Test::Methods
+
+  let(:exporter) { EXPORTER }
+  let(:spans) { exporter.finished_spans }
+  let(:span) { exporter.finished_spans.last }
+  let(:rails_app) { DEFAULT_RAILS_APP }
+
+  # Clear captured spans
+  before { exporter.reset }
+
+  it 'sets the view runtime' do
+    get '/ok'
+
+    _(last_response.body).must_equal 'actually ok'
+    _(last_response.ok?).must_equal true
+
+    _(span.instrumentation_library.name).must_equal 'OpenTelemetry::Instrumentation::Rack'
+    _(span.instrumentation_library.version).must_equal OpenTelemetry::Instrumentation::Rack::VERSION
+
+    _(span.attributes['action_controller.view_runtime_ms']).must_be :>=, 0
+  end
+
+  it 'sets the db runtime' do
+    get '/query'
+
+    _(last_response.body).must_equal 'make query'
+    _(last_response.ok?).must_equal true
+
+    _(span.instrumentation_library.name).must_equal 'OpenTelemetry::Instrumentation::Rack'
+    _(span.instrumentation_library.version).must_equal OpenTelemetry::Instrumentation::Rack::VERSION
+
+    _(span.attributes['action_controller.db_runtime_ms']).must_be :>=, 0
+  end
+
+  it 'sets common active support notification attributes' do
+    get '/ok'
+
+    _(last_response.body).must_equal 'actually ok'
+    _(last_response.ok?).must_equal true
+
+    _(span.instrumentation_library.name).must_equal 'OpenTelemetry::Instrumentation::Rack'
+    _(span.instrumentation_library.version).must_equal OpenTelemetry::Instrumentation::Rack::VERSION
+
+    below_rails('6') do
```

Comment:
Unsure how best to writing tests for different rails versions. Appraisal includes rails 5.2 and this allocation and cpu time was only added in 6.0